File: kotlin/udemy/src/_03_Types/TypeCastingBasic.java
package _03_Types;

import java.util.Random;

public class TypeCastingBasic {
    public static void main(String[] args) {
        Object result;

        if (new Random().nextBoolean()) {
            result = "Hello";
        } else {
            result = 12345;
        }

        System.out.println("Result is " + result);

        if (result instanceof String) {
            System.out.println(((String) result).substring(0, 1));
        } else {
            System.out.println(((Integer) result) + 3);
        }
    }
}

File: kotlin/udemy/src/_03_Types/TypeCastingBasic.kt
package _03_Types

import java.util.*

fun main() {
    val result: Any

    // 일반적인 언어의 삼항연산자와는 다르다
    result = if (Random().nextBoolean()) "Hello" else 1234

    println("Result is $result")

    if (result is String) {
        println(result.substring(0, 1)) // Smart Casting
    } else {
        println(result as Int + 3) // casting은 `as`로 한다
    }
}
